src/picongpu/include/particles/ionization/byCollision/ThomasFermi/ThomasFermi_Impl.hpp

```diff
@@ -0,0 +1,274 @@
+/**
+ * Copyright 2016-2017 Marco Garten
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <boost/type_traits/integral_constant.hpp>
+
+#include "simulation_defines.hpp"
+#include "traits/Resolve.hpp"
+#include "traits/UsesRNG.hpp"
+#include "mappings/kernel/AreaMapping.hpp"
+
+#include "fields/FieldTmp.hpp"
+
+#include "particles/ionization/byCollision/ThomasFermi/ThomasFermi.def"
+#include "particles/ionization/byCollision/ThomasFermi/AlgorithmThomasFermi.hpp"
+#include "particles/ionization/ionization.hpp"
+
+#include "compileTime/conversion/TypeToPointerPair.hpp"
+#include "memory/boxes/DataBox.hpp"
+
+#include "particles/ionization/ionizationMethods.hpp"
+
+#include "random/methods/XorMin.hpp"
+#include "random/distributions/Uniform.hpp"
+#include "random/RNGProvider.hpp"
+
+namespace picongpu
+{
+namespace traits
+{
+    /** specialization of the UsesRNG trait
+     * --> ionization module uses random number generation
+     */
+    template<typename T_IonizationAlgorithm, typename T_DestSpecies, typename T_SrcSpecies>
+    struct UsesRNG<particles::ionization::ThomasFermi_Impl<T_IonizationAlgorithm, T_DestSpecies, T_SrcSpecies> > :
+    public boost::true_type
+    {
+    };
+} // namespace traits
+
+namespace particles
+{
+namespace ionization
+{
+
+    /** \struct ThomasFermi_Impl
+     *
+     * \brief Thomas-Fermi pressure ionization - Implementation
+     *
+     * \tparam T_DestSpecies electron species to be created
+     * \tparam T_SrcSpecies particle species that is ionized
+     */
+    template<typename T_IonizationAlgorithm, typename T_DestSpecies, typename T_SrcSpecies>
+    struct ThomasFermi_Impl
+    {
+
+        typedef T_DestSpecies DestSpecies;
```

Comment:
if you don't mind, please use the C++11 `using NewType = OtherType;` since it's more readable for new code.